### PR TITLE
Add events property

### PR DESCRIPTION
**Pull request by vrunoa**
Get the session to get events timing information. @KazuCocoa @jlipps let me know what you guys think on this approach. 

I'm seeing the branch name now "porperty" 🤦🏻‍♂️

### CODE REVIEW

File: appium/webdriver/webdriver.py [hunk 1]
```diff
@@ -691,6 +691,24 @@ def battery_info(self):
         """
         return self.execute_script('mobile: batteryInfo')
 
+    @property
```
Comment (KazuCocoa) [inline appium/webdriver/webdriver.py]: What about like below? The naming is from https://github.com/appium/appium/blob/acb88dca744fadf1e48ede0af6c1bdaa742e6c14/commands-yml/commands/session/get.yml . We can update the `self.driver.desired_capabilities()` to `self.driver.session_capabilities`.

```python
@property
def session_capabilities(self):
    return self.execute(Command.GET_SESSION)["value"]

@property
def events(self):
    try:
        self.session_capabilities["events"]
    except....
```

File: test/unit/webdriver/webdriver_test.py [hunk 2]
```diff
@@ -255,6 +256,35 @@ def test_create_session_register_uridirect_no_direct_connect_path(self):
         assert 'http://localhost:4723/wd/hub' == driver.command_executor._url
         assert ['NATIVE_APP', 'CHROMIUM'] == driver.contexts
 
+    @httpretty.activate
+    def test_get_events(self):
+        driver = ios_w3c_driver()
```
Comment (KazuCocoa) [inline test/unit/webdriver/webdriver_test.py]: 👍

File: appium/webdriver/webdriver.py [hunk 3]
```diff
@@ -691,6 +691,24 @@ def battery_info(self):
...
+    def events(self):
+        """ Retrieves event timing information from the current session
+        Usage:
+            events = driver.events
+
+        Returns:
+            `dict containing event timimgs information from the current session`
```
Comment (ki4070ma) [inline appium/webdriver/webdriver.py]: typo: timimgs -> timings

And you can describe like below.

`dict: containing event timings information from the current session`

e.g. 
https://github.com/appium/python-client/blob/master/appium/webdriver/extensions/screen_record.py#L103-L106

File: appium/webdriver/webdriver.py [hunk 4]
```diff
@@ -691,6 +691,24 @@ def battery_info(self):
...
+        """
+        self.command_executor._commands[Command.GET_SESSION] = \
+            ('GET', '/session/$sessionId')
+        session = self.execute(Command.GET_SESSION)
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: it makes sense to have a separate method for getSession call

File: appium/webdriver/webdriver.py (hunk 1 above)
Comment (vrunoa) [reply appium/webdriver/webdriver.py]: Yeap, this looks like a much better approach.

File: appium/webdriver/webdriver.py [hunk 5]
```diff
@@ -691,6 +691,24 @@ def battery_info(self):
...
+    @property
+    def events(self):
+        """ Retrieves event timing information from the current session
+        Usage:
+            events = driver.events
+
+        Returns:
+            `dict containing event timimgs information from the current session`
+        """
+        self.command_executor._commands[Command.GET_SESSION] = \
```
[the remark this answers is not in the document]
Comment (vrunoa) [reply appium/webdriver/webdriver.py]: Done!

File: appium/webdriver/webdriver.py [hunk 6]
```diff
@@ -691,6 +691,32 @@ def battery_info(self):
         """
         return self.execute_script('mobile: batteryInfo')
 
+    @property
+    def session_capabilities(self):
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: how about having `session` name?

File: appium/webdriver/webdriver.py [hunk 7]
```diff
@@ -691,6 +691,32 @@ def battery_info(self):
...
+        """ Retrieves events information from the current session
+        Usage:
+            events = driver.events
+
+        Returns:
+            `dict containing events timing information from the current session`
+        """
+        try:
+            session_capabilities = self.session_capabilities
+            return session_capabilities["events"]
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: Please use single quotes everywhere

File: appium/webdriver/webdriver.py [hunk 8]
```diff
@@ -691,6 +691,32 @@ def battery_info(self):
...
+            events = driver.events
+
+        Returns:
+            `dict containing events timing information from the current session`
+        """
+        try:
+            session_capabilities = self.session_capabilities
+            return session_capabilities['events']
+        except Exception:
+            logger.warning('Could not find events information in the session. Session: {}'.format(session_capabilities))
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: is it expected to mute exceptions? How we can then distinguish "no events" from "cannot get any events"?

**Comment by ki4070ma** [comment]
Retriggered functional android test on azure pipelines

* [Failure case] test_toggle_location_services

Error log
```
selenium.common.exceptions.WebDriverException: Message:
An unknown server-side error occurred while processing the command.
Original error: Error executing adbExec.
Original error: 'Command '/Users/vsts/Library/Android/sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pm install /data/local/tmp/appium_cache/bc38a04068b79fe3b459303c28bb4b7103bfb6c9.apk' timed out after 20000ms'.
Try to increase the 20000ms adb execution timeout represented by 'uiautomator2ServerInstallTimeout' capability
```

https://github.com/appium/python-client/pull/413 will fix above error.

**Comment by ki4070ma** [comment]
Link to https://github.com/appium/python-client/pull/427

File: appium/webdriver/webdriver.py (hunk 8 above)
Comment (vrunoa) [reply appium/webdriver/webdriver.py]: I'll log the exception so we have more visibility on the error.

File: appium/webdriver/webdriver.py (hunk 6 above)
Comment (vrunoa) [reply appium/webdriver/webdriver.py]: yeah, I think it make more sense to call it session.

File: appium/webdriver/webdriver.py [hunk 9]
```diff
@@ -691,6 +691,32 @@ def battery_info(self):
         """
         return self.execute_script('mobile: batteryInfo')
 
+    @property
+    def session(self):
+        """ Retrieves session information from the current session
+        Usage:
+            session_capabilities = driver.session_capabilities
```
Comment (mykola-mokhnach) [inline appium/webdriver/webdriver.py]: please update

**Comment by dpgraham** [comment]
@ki4070ma @KazuCocoa can we merge this and publish it?

**Comment by dpgraham** [comment]
@vrunoa is this still just a WIP?

**Comment by vrunoa** [comment]
Ready for release 🚀

**Comment by KazuCocoa** [comment]
waiting for CI

**Comment by ki4070ma** [comment]
Need to.do https://github.com/appium/python-client/pull/427#pullrequestreview-276884246

**Comment by KazuCocoa** [comment]
mm, next is 'python -m isort -rc .'

**Pull request closed without merging**